### DISCUSSION TITLE
What is the recommended way to map from XID8 datatype?

### DISCUSSION THREAD

**Discussion by podjay** [jackc/pgx#2137] [Q&A] [answered]
I have created a table in which I store an XID8 datatype. Is there a sensible `pgtype` that I can use to map from this column?

I see that there is a `pgtype.XID` which will map up to `MaxUint32` which I'm sure will be fine (for some time) but I expected, perhaps, a `pgtype.XID8`. 

For the people using/storing XID8, how are you reading and interpreting this using pgx?

Thanks!

**Comment by jackc** [comment, marked as answer]
At the moment there is no support for xid8. You would have to scan it to a string and parse it yourself or cast it to a bigint in PostgreSQL.

However, I just made a PR to add support for unsigned 64 bit integers like xid8. https://github.com/jackc/pgx/pull/2142. It's basically just find and replace of the existing uint32 code. See if that works for you.

**Reply by podjay** [reply]
Hey @jackc; thank you for the insight and quick implementation. I gave the `pgx/v5` version a whirl locally in my project (we currently have some pgx/v4 types) and I didn't have great success when querying the raw XID8 value - the uint64 scanner seemed always return `cannot scan []uint8`. 

I also noticed that I was unable to cast to `bigint` - my version of postgres (15.8) didn't seem to like it; I got an error like: `pq: cannot cast type xid8 to bigint` so presumably that is not an option.

However, I'll do some more investigation on casting xid8::text and back because that looks like it will get me quite far.

Thanks again!
